### PR TITLE
RavenDB-11107 Give more clear error message when failing to run the server

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: src/Raven.Server/Program.cs [hunk 1]
```diff
@@ -187,9 +187,17 @@ public static int Main(string[] args)
                         }
                         catch (Exception e)
                         {
+                            var message =
```
Comment (ppekrol) [inline src/Raven.Server/Program.cs]: We need to detect that this was a port binding exception.

File: src/Raven.Server/Program.cs [hunk 2]
```diff
@@ -187,9 +187,17 @@ public static int Main(string[] args)
...
+                                $"{e.Message}{Environment.NewLine}Port might be already in use.{Environment.NewLine}Try running with an unused port.{Environment.NewLine}" +
+                                $"You can change the port using one of the following options:{Environment.NewLine}" +
+                                $"1) Run the server from the command line with --ServerUrl option.{Environment.NewLine}" +
+                                $"2) Change the ServerUrl property in setting.json file.{Environment.NewLine}" +
```
Comment (ppekrol) [inline src/Raven.Server/Program.cs]: This should be swapped with 1)

File: src/Raven.Server/Program.cs [hunk 3]
```diff
@@ -187,9 +187,17 @@ public static int Main(string[] args)
...
+                                $"3) Add RAVEN_ServerUrl in linux or RAVEN.ServerUrl in windows to the Environment Variables.{Environment.NewLine}" +
```
Comment (ppekrol) [inline src/Raven.Server/Program.cs]: linux -> Linux
windows -> Windows

using `RAVEN_ServerUrl` on both platforms is sufficient.

File: src/Raven.Server/Program.cs [hunk 4]
```diff
@@ -187,9 +187,17 @@ public static int Main(string[] args)
...
+                                "For more information go to https://ravendb.net/docs/article-page/4.0/csharp/server/configuration/configuration-options";
```
Comment (ppekrol) [inline src/Raven.Server/Program.cs]: Talk with Tom about this. He will generate a link for you.

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Program.cs (hunk 3 above)
Comment (IdanHaim) [reply src/Raven.Server/Program.cs]: Do we want to cover both ways?

File: src/Raven.Server/Program.cs (hunk 2 above)
Comment (IdanHaim) [reply src/Raven.Server/Program.cs]: Running a server from the command line will use the args from the command line and not the Setting.json file

File: src/Raven.Server/Program.cs (hunk 4 above)
Comment (IdanHaim) [reply src/Raven.Server/Program.cs]: Done

File: src/Raven.Server/Program.cs (hunk 1 above)
Comment (IdanHaim) [reply src/Raven.Server/Program.cs]: We can detect this by checking the InnerException e.InnerException will be AddressInUseException

File: src/Raven.Server/Program.cs (hunk 3 above)
Comment (IdanHaim) [reply src/Raven.Server/Program.cs]: Going to keep this

File: src/Raven.Server/Program.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Program.cs]: On Linux as well?
What about not permission for low ports on Linux?

**Pull request merged**